src/dicom-file.c

```diff
@@ -1296,11 +1297,19 @@ bool dcm_filehandle_prepare_read_frame(DcmError **error,
                     return false;
                 }
             } else {
+                if (filehandle->desc.bits_allocated == 1) {
+                    dcm_error_set(error, DCM_ERROR_CODE_PARSE,
+                                  "reading frame item failed",
+                                  "1-bit pixel storage (Bits Allocated == 1) not supported");
+                    return false;
+                }
                 for (uint32_t i = 0; i < filehandle->num_frames; i++) {
-                    filehandle->offset_table[i] = i *
-                                                  filehandle->desc.rows *
-                                                  filehandle->desc.columns *
-                                                  filehandle->desc.samples_per_pixel;
+                    int64_t bits_allocated = (int64_t)filehandle->desc.rows *
+                                             (int64_t)filehandle->desc.columns *
+                                             (int64_t)filehandle->desc.bits_allocated *
```

Comment:
You could change this to `filehandle->desc.bits_allocated / 8` and you wouldn't need the intermediate `uint64_t` value.

Reply:
Thanks, this was my original approach, and will work fine for values of 8 and above. I was thinking about 1-bit support but don’t have a personal need for that currently and it seems a bit niche. I may revisit in the future if it becomes an issue.